packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -21,8 +23,41 @@ class UserSessionManager implements IUserSessionManager {
     this.storage = storage;
   }
 
-  setStorage(storage: IStore) {
+  init(storage: IStore) {
     this.storage = storage;
+
+    this.setUserId(this.getUserId() || '');
+    this.setUserTraits(this.getUserTraits() || {});
+    this.setGroupId(this.getGroupId() || '');
+    this.setGroupTraits(this.getGroupTraits() || {});
+    this.setAnonymousId(this.getAnonymousId());
+
+    const initialReferrer = this.getInitialReferrer();
+    const initialReferringDomain = this.getInitialReferringDomain();
+
+    state.session.rl_page_init_referrer.value =
+      initialReferrer !== null ? initialReferrer : undefined;
+    state.session.rl_page_init_referring_domain.value =
+      initialReferringDomain !== null ? initialReferringDomain : undefined;
+
+    if (initialReferrer === null && initialReferringDomain === null) {
+      const referrer = getReferrer();
+      this.setInitialReferrer(referrer);
+      this.setInitialReferringDomain(referrer);
```

Comment:
Why are we setting referrer as domain here?

Reply:
You missed extracting the domain first.

Reply:
Can you point me to that?

Reply:
https://github.com/rudderlabs/rudder-sdk-js/blob/f085febde59ea74d64927a112526e7e51d5fa2b4/src/core/analytics.js#L135

Reply:
Done